zerver/data_import/slack.py

```diff
@@ -999,6 +999,10 @@ def channel_message_to_zerver_message(
                 topic_name = f"{thread_date} Slack thread {count}"
                 thread_map[thread_ts_str] = topic_name
 
+        # The topic name is empty string for DMs and GDMs.
+        if is_private:
+            topic_name = ""
```

Comment:
Is the scenario where this might happen a Slack thread in a DM?

Reply:
> Is the scenario where this might happen a Slack thread in a DM?

I would say that the scenario where this might happen is either when the message is a DM, GDM or a thread in a DM. I understand it as if a thread in a DM should be "flattened" into the main conversation in Zulip. Did I understand your question correctly? Do you think I have the wrong approach?

Reply:
Yes, this would cover DM, group DM, and thread in group DM (this will have an additional `thread_ts` field)

Reply:
Could this logic be moved to `build_message`, so that it's done correctly for all data import tools, including future ones we might write?

Reply:
I could potentially add something like this to build_message: 
`  recipient = Recipient.objects.get(id=recipient_id)
    # Now you can access the type_id from the recipient object
    recipient_type_id = recipient.type_id
    if recipient_type_id == 1 or recipient_type_id == 3:
        topic_name = ""`
what do you think?

Reply:
I'm a little confused because we shouldn't be doing any writes to the database from these tools, so `Recipient.objects.get(id=recipient_id)` doesn't make any sense. 

But I think `build_message` could reasonably clear the `topic` field if it's not the `""` on DMs, right? The caller should be passing something that clearly indicates that, or could easily do so if it isn't.

Reply:
Yes, we could likely clear the `topic_name` in `build_message`. However, the reason for my solution is that I couldn’t find anything in build_message that indicates whether the message is a DM. Perhaps we could add a field like `is_private` to make it possible to clear the topic in `build_message`. What do you think?

Reply:
Adding a parameter seems fine, yes.